can/interfaces/ixxat/canlib_vcinpl2.py

```diff
@@ -0,0 +1,920 @@
+                                    self._message.abData[0]
+                                ),
+                            )
+                        )
+
+                    elif (
+                        self._message.uMsgInfo.Bits.type == constants.CAN_MSGTYPE_ERROR
+                    ):
+                        log.warning(
+                            CAN_ERROR_MESSAGES.get(
+                                self._message.abData[0],
+                                "Unknown CAN error message code {}".format(
+                                    self._message.abData[0]
+                                ),
+                            )
+                        )
+
+                    elif (
+                        self._message.uMsgInfo.Bits.type == constants.CAN_MSGTYPE_STATUS
+                    ):
+                        log.info(_format_can_status(self._message.abData[0]))
+                        if self._message.abData[0] & constants.CAN_STATUS_BUSOFF:
+                            raise VCIBusOffError()
+
+                    elif (
+                        self._message.uMsgInfo.Bits.type
+                        == constants.CAN_MSGTYPE_TIMEOVR
+                    ):
+                        pass
+                    else:
+                        log.warning("Unexpected message info type")
+
+                if t0 is not None:
+                    remaining_ms = timeout_ms - int((_timer_function() - t0) * 1000)
+                    if remaining_ms < 0:
+                        break
+
+        if not data_received:
+            # Timed out / can message type is not DATA
+            return None, True
+
+        data_len = can.util.dlc2len(self._message.uMsgInfo.Bits.dlc)
+        # The _message.dwTime is a 32bit tick value and will overrun,
+        # so expect to see the value restarting from 0
+        rx_msg = Message(
+            timestamp=self._message.dwTime
+            / self._tick_resolution,  # Relative time in s
+            is_remote_frame=bool(self._message.uMsgInfo.Bits.rtr),
+            is_extended_id=bool(self._message.uMsgInfo.Bits.ext),
+            arbitration_id=self._message.dwMsgId,
+            dlc=data_len,
+            data=self._message.abData[:data_len],
+            channel=self.channel,
+        )
+
+        return rx_msg, True
+
+    def send(self, msg: Message, timeout: Optional[float] = None) -> None:
+        """
+        Sends a message on the bus. The interface may buffer the message.
+
+        :param msg:
+            The message to send.
+        :param timeout:
+            Timeout after some time.
+        :raise:
+            :class:CanTimeoutError
+            :class:CanOperationError
+        """
+        # This system is not designed to be very efficient
+        message = structures.CANMSG2()
+        message.uMsgInfo.Bits.type = constants.CAN_MSGTYPE_DATA
+        message.uMsgInfo.Bits.rtr = 1 if msg.is_remote_frame else 0
```

Comment:
Maybe initialize `esi` - not sure though.

Reply:
I agree, also changed.